extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -177,6 +179,10 @@ export class BookTocManager implements IBookTocManager {
 				newToc[index] = newSection;
 			}
 		}
+		if (!this._foundSection) {
+			// if the section was not found, throw an error and restore both toc files.
+			throw (new Error(loc.sectionNotFound(findSection.title)));
```

Comment:
Given that the section needs to have more than just the title in common (it also needs to have either the same file or the same URL) I would suggest also including that information in the error message.